icechunk-python/tests/test_dask.py

```diff
@@ -0,0 +1,22 @@
+import pytest
+
+pytest.importorskip("distributed")
+from .test_xarray import create_test_data, roundtrip
+from distributed import Client
+from distributed.utils_test import (  # noqa
+    # need to import all these fixtures
+    cluster,
+    loop,
+    loop_in_thread,
+    cleanup,
+)
+
+from xarray.testing import assert_identical
+
+
+def test_xarray_to_icechunk_distributed(loop):
```

Comment:
@phofl I'm having trouble with a distributed test.

1. Apparently the `client` fixture runs twice. Why?
2. The first one succeeds, but the second fails with this cleanup error. 

Is there another testing interface I should be using?

```
_____________________________________________________ ERROR at teardown of test_xarray_to_icechunk_distributed ______________________________________________________
1 thread(s) were leaked from test

------ Call stack of leaked thread 1/1: <Thread(zarr_io, started daemon 6176387072)> ------
  File "/Users/deepak/miniforge3/envs/icechunk/lib/python3.12/threading.py", line 1032, in _bootstrap
	self._bootstrap_inner()
  File "/Users/deepak/miniforge3/envs/icechunk/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
	self.run()
  File "/Users/deepak/miniforge3/envs/icechunk/lib/python3.12/threading.py", line 1012, in run
	self._target(*self._args, **self._kwargs)
  File "/Users/deepak/miniforge3/envs/icechunk/lib/python3.12/asyncio/base_events.py", line 641, in run_forever
	self._run_once()
  File "/Users/deepak/miniforge3/envs/icechunk/lib/python3.12/asyncio/base_events.py", line 1948, in _run_once
	event_list = self._selector.select(timeout)
  File "/Users/deepak/miniforge3/envs/icechunk/lib/python3.12/selectors.py", line 566, in select
	kev_list = self._selector.control(None, max_ev, timeout)
----------------------------------------------------------------------- Captured stderr call ------------------------------------------------------------------------
2024-11-14 16:11:00,259 - distributed.scheduler - INFO - State start
2024-11-14 16:11:00,261 - distributed.scheduler - INFO -   Scheduler at:     tcp://127.0.0.1:61234
2024-11-14 16:11:00,261 - distributed.scheduler - INFO -   dashboard at:  http://127.0.0.1:8787/status
2024-11-14 16:11:00,261 - distributed.scheduler - INFO - Registering Worker plugin shuffle
2024-11-14 16:11:00,272 - distributed.worker - INFO -       Start worker at:      tcp://127.0.0.1:61236
2024-11-14 16:11:00,272 - distributed.worker - INFO -       Start worker at:      tcp://127.0.0.1:61235
2024-11-14 16:11:00,273 - distributed.worker - INFO -          Listening to:      tcp://127.0.0.1:61236
2024-11-14 16:11:00,273 - distributed.worker - INFO -          Listening to:      tcp://127.0.0.1:61235
2024-11-14 16:11:00,273 - distributed.worker - INFO -          dashboard at:            127.0.0.1:61238
2024-11-14 16:11:00,273 - distributed.worker - INFO -          dashboard at:            127.0.0.1:61237
2024-11-14 16:11:00,273 - distributed.worker - INFO - Waiting to connect to:      tcp://127.0.0.1:61234
2024-11-14 16:11:00,273 - distributed.worker - INFO - Waiting to connect to:      tcp://127.0.0.1:61234
2024-11-14 16:11:00,273 - distributed.worker - INFO - -------------------------------------------------
2024-11-14 16:11:00,273 - distributed.worker - INFO - -------------------------------------------------
2024-11-14 16:11:00,273 - distributed.worker - INFO -               Threads:                          1
2024-11-14 16:11:00,273 - distributed.worker - INFO -               Threads:                          1
2024-11-14 16:11:00,273 - distributed.worker - INFO -                Memory:                  24.00 GiB
2024-11-14 16:11:00,273 - distributed.worker - INFO -                Memory:                  24.00 GiB
2024-11-14 16:11:00,273 - distributed.worker - INFO -       Local Directory: /var/folders/sc/hsdthq2x7mnbfvwnk7rp4xmr0000gn/T/dask-scratch-space/worker-j0mr1vgr
2024-11-14 16:11:00,273 - distributed.worker - INFO -       Local Directory: /var/folders/sc/hsdthq2x7mnbfvwnk7rp4xmr0000gn/T/dask-scratch-space/worker-yq0pvjqx
2024-11-14 16:11:00,273 - distributed.worker - INFO - -------------------------------------------------
2024-11-14 16:11:00,273 - distributed.worker - INFO - -------------------------------------------------
2024-11-14 16:11:00,409 - distributed.scheduler - INFO - Register worker <WorkerState 'tcp://127.0.0.1:61235', status: init, memory: 0, processing: 0>
2024-11-14 16:11:00,545 - distributed.scheduler - INFO - Starting worker compute stream, tcp://127.0.0.1:61235
2024-11-14 16:11:00,545 - distributed.core - INFO - Starting established connection to tcp://127.0.0.1:61240
2024-11-14 16:11:00,545 - distributed.scheduler - INFO - Register worker <WorkerState 'tcp://127.0.0.1:61236', status: init, memory: 0, processing: 0>
2024-11-14 16:11:00,545 - distributed.scheduler - INFO - Starting worker compute stream, tcp://127.0.0.1:61236
2024-11-14 16:11:00,545 - distributed.worker - INFO - Starting Worker plugin shuffle
2024-11-14 16:11:00,545 - distributed.core - INFO - Starting established connection to tcp://127.0.0.1:61239
2024-11-14 16:11:00,546 - distributed.worker - INFO -         Registered to:      tcp://127.0.0.1:61234
2024-11-14 16:11:00,546 - distributed.worker - INFO - -------------------------------------------------
2024-11-14 16:11:00,546 - distributed.worker - INFO - Starting Worker plugin shuffle
2024-11-14 16:11:00,546 - distributed.core - INFO - Starting established connection to tcp://127.0.0.1:61234
2024-11-14 16:11:00,546 - distributed.worker - INFO -         Registered to:      tcp://127.0.0.1:61234
2024-11-14 16:11:00,546 - distributed.worker - INFO - -------------------------------------------------
2024-11-14 16:11:00,547 - distributed.core - INFO - Starting established connection to tcp://127.0.0.1:61234
2024-11-14 16:11:00,556 - distributed.scheduler - INFO - Receive client connection: Client-b6862d92-a2dd-11ef-b8ef-22fa3d08abc4
2024-11-14 16:11:00,556 - distributed.core - INFO - Starting established connection to tcp://127.0.0.1:61259
2024-11-14 16:11:01,060 - distributed.scheduler - INFO - Remove client Client-b6862d92-a2dd-11ef-b8ef-22fa3d08abc4
2024-11-14 16:11:01,061 - distributed.core - INFO - Received 'close-stream' from tcp://127.0.0.1:61259; closing.
2024-11-14 16:11:01,061 - distributed.scheduler - INFO - Remove client Client-b6862d92-a2dd-11ef-b8ef-22fa3d08abc4
2024-11-14 16:11:01,061 - distributed.scheduler - INFO - Close client connection: Client-b6862d92-a2dd-11ef-b8ef-22fa3d08abc4
====================================================================== short test summary info ======================================================================
ERROR test_dask.py::test_xarray_to_icechunk_distributed - Failed: 1 thread(s) were leaked from test
==================================================================== 1 passed, 1 error in 6.75s ================
```

Reply:
I have never used the loop fixture tbh, we normally use the ``@gen_cluster`` decorator for all tests these days

This would look like:

```
from distributed.utils_test import gen_cluster

@gen_cluster(client=True)
async def test_something(c, s, *ws):
    # do stuff
    pass
```